cmake/TargetArch.cmake

```diff
@@ -146,6 +146,15 @@ function(target_architecture output_var)
       # Get rid of the value marker leaving just the architecture name
       string(REPLACE "cmake_ARCH " "" ARCH "${ARCH}")
 
+      # If compiling for riscv64, automatically bake in Boost.Context
+      if(ARCH STREQUAL "riscv64")
+        hpx_set_option(
+            HPX_WITH_GENERIC_CONTEXT_COROUTINES
+            VALUE ON
+            FORCE
+        )
+      endif()
```

Comment:
May I suggest adding this option here instead: https://github.com/STEllAR-GROUP/hpx/blob/d26d4126260d93ff9a4ddad877e7ee27c115ed9e/CMakeLists.txt#L1564-L1577?

Reply:
all set! had to move target arch detection a little earlier.

Reply:
Thanks!